### PR TITLE
Add assertion to test equality of sequences.

### PR DESCRIPTION
**Pull request by amirkamil**

### CODE REVIEW

File: unit_test_framework.h [hunk 1]
```diff
@@ -327,10 +327,12 @@ void assert_not_equal(First&& first, Second&& second, int line_number) {
 
 template <typename First, typename Second>
 void assert_sequence_equal(First&& first, Second&& second, int line_number) {
-    auto it1 = std::begin(first);
-    auto it2 = std::begin(second);
-    auto end1 = std::end(first);
-    auto end2 = std::end(second);
+    using std::begin;
```
Comment (james-perretta) [inline unit_test_framework.h]: Can you explain this change to me?

Comment (amirkamil) [reply unit_test_framework.h]: https://en.cppreference.com/w/cpp/language/adl#Notes

Comment (james-perretta) [reply unit_test_framework.h]: So there are other specializations of begin() and end() that need to be considered? I can follow up with you about this some other time.

Comment (amirkamil) [reply unit_test_framework.h]: There are a couple mentioned [here](https://en.cppreference.com/w/cpp/iterator/begin#User-defined_overloads), neither of which is important for us. But I could see this showing up in a third-party container library.

**Pull request merged**